andela/wolfsbane-backend
# #167972760 Setup and Integrate Transactional Email Service with Sendgrid

#### Description of Task to be completed?
- User should be able to receive transactional email, user activities such as account verification, request for password recovery, trip request should trigger transactional email in order to aid user's experience

 #### How should this be manually tested?
- run ```git clone https://github.com/andela/wolfsbane-backend.git```
- cd into the cloned directory
- run ```git fetch origin pull/12/head:ft-integrate-email-service-167972760```
- run ```npm install``` : to install packages and ensure the @sendgrid and mocha libraries are installed
- Ensure your internet connection is stable
- Replace 'firstName' string from src/test/fakeData/fakeData.js file from with desired First Name
- Replace 'xxxxxx' string from src/test/fakeData/fakeData.js file with desired token
- Replace 'appropriatemail@gmail.com' string from src/test/fakeData/fakeData.js file with desired email
- The email category variable has been set for various tests already
- npm ```run test``` : to run mock test from src/test/services/autoMailer.test.js  and src/test/utils/emailTemplatesFunction.test.js

 #### Any background context you want to provide?
- To employ this service, import/ destructure senMail from services/index
- Ensure to have the SENDGRID_API_KEY variable set in your environment
- Ensure to pass recipient's email, category of email and provisional data to the email service function e.g: sendMail(email, confirmAccount, [firstName, token]);
- Email could be a string or array, category of email for now could be either of 'confirmAccount', 'passwordRecovery', or tripRequest, and the provisional data of course an array

 #### What are the relevant pivotal tracker stories?
[#167972760](https://www.pivotaltracker.com/story/show/167972760)

 #### Screenshots (if appropriate)
![001](https://user-images.githubusercontent.com/42122976/64324596-90b49380-cfbe-11e9-9bb3-e52d358f30ec.png)
![002](https://user-images.githubusercontent.com/42122976/64324601-914d2a00-cfbe-11e9-875a-2fd8e3a3fc6f.png)
![new trip request](https://user-images.githubusercontent.com/42122976/64324405-3ddadc00-cfbe-11e9-9fbb-84a2c0881d23.png)
![feedback101](https://user-images.githubusercontent.com/42122976/64324416-46331700-cfbe-11e9-9388-304bd35e6df1.png)


 #### Questions:
